src/Devices/Ethernet_TC.cpp

```diff
@@ -0,0 +1,32 @@
+#include "Ethernet_TC.h"
+using namespace std;
+
+TC_ethernet *TC_ethernet::_instance = nullptr;
+
+TC_ethernet::TC_ethernet() {
+  pinMode(4, OUTPUT);
+  digitalWrite(4, HIGH);
+  if (Ethernet.begin(mac) == 0) {
```

Comment:
If the `begin()` method is successful, does `IP` get initialized?

Reply:
The IP is either initialized through DHCP in the `begin(mac)` contructor, or (I moved the constructor so it won't be declared unless needed) it will be given a defaultIP with the `begin(mac, IP)` constructor